instrumentation/opentelemetry-instrumentation-requests/tests/test_requests_integration.py

```diff
@@ -281,27 +243,93 @@ def test_requests_exception_with_response(self, *_, **__):
             span.status.canonical_code, StatusCanonicalCode.INTERNAL
         )
 
-    @mock.patch("requests.Session.send", side_effect=Exception)
+    @mock.patch("requests.adapters.HTTPAdapter.send", side_effect=Exception)
     def test_requests_basic_exception(self, *_, **__):
-
         with self.assertRaises(Exception):
-            requests.get(self.URL)
+            self.perform_request(self.URL)
 
-        span_list = self.memory_exporter.get_finished_spans()
-        self.assertEqual(len(span_list), 1)
+        span = self.assert_span()
         self.assertEqual(
-            span_list[0].status.canonical_code, StatusCanonicalCode.UNKNOWN
+            span.status.canonical_code, StatusCanonicalCode.UNKNOWN
         )
 
-    @mock.patch("requests.Session.send", side_effect=requests.Timeout)
+    @mock.patch(
+        "requests.adapters.HTTPAdapter.send", side_effect=requests.Timeout
+    )
     def test_requests_timeout_exception(self, *_, **__):
-
         with self.assertRaises(Exception):
-            requests.get(self.URL)
+            self.perform_request(self.URL)
+
+        span = self.assert_span()
+        self.assertEqual(
+            span.status.canonical_code, StatusCanonicalCode.DEADLINE_EXCEEDED
+        )
+
+
+def setup_test(url: str):
+    RequestsInstrumentor().instrument()
+    httpretty.enable()
+    httpretty.register_uri(httpretty.GET, url, body="Hello!")
+
+
+def teardown_test():
+    RequestsInstrumentor().uninstrument()
+    httpretty.disable()
+
+
+class TestRequestsIntegration(TestBase, RequestsIntegrationTestBase):
```

Comment:
I believe you could actually incorporate this setup and teardown code into RequestsIntegrationTestBase.

abc.ABC is okay to use as a second parent class, and will still have the abstractmethod enforcing logic as if it was the first class passed.

Then you can just subclass everything from TestRequestsBase as before.